python/pyarrow/_parquet.pyx

```diff
@@ -1767,7 +1767,7 @@ cdef class ParquetWriter(_Weakrefable):
             int64_t c_row_group_size
 
         if row_group_size is None or row_group_size == -1:
-            c_row_group_size = ctable.num_rows()
+            c_row_group_size = min(ctable.num_rows(), 1024*1024)
```

Comment:
should we declare a constant rather than 1024*1024 directly?

Reply:
I'll defer to @jorisvandenbossche only because I don't actually know what a constant should like like in this file (style wise).  I can't find any good examples.

Reply:
Either way is fine for me, but so it would like the following (defined at the top of the file, after the imports):

https://github.com/apache/arrow/blob/2ce4a38fcf1fbd0d759b6c2cdf657f27ee6f02a9/python/pyarrow/_dataset.pyx#L42-L46

(but for something that is not reused multiple times, it is less worth it, I think)

Reply:
Ok,I have to fix a lint issue anyways so I'll add a constant real quick for readability.